pkg/service/service_manager.go

```diff
@@ -0,0 +1,128 @@
+package service
+
+import (
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+
+	"github.com/k8snetworkplumbingwg/sriov-network-operator/pkg/utils"
+)
+
+type ServiceManager interface {
+	ReadService(string) (*Service, error)
+	EnableService(service *Service) error
+	DisableService(service *Service) error
+	StartService(service string) error
+	StopService(service string) error
+}
+
+type serviceManager struct {
+	chroot string
+}
+
+func NewServiceManager(chroot string) ServiceManager {
+	root := chroot
+	if root == "" {
+		root = "/"
+	}
+	return &serviceManager{root}
+}
+
+// ReadService read service from given path
+func (sm *serviceManager) ReadService(servicePath string) (*Service, error) {
+	data, err := ioutil.ReadFile(path.Join(sm.chroot, servicePath))
+	if err != nil {
+		return nil, err
+	}
+
+	return &Service{
+		Name:    filepath.Base(servicePath),
+		Path:    servicePath,
+		Content: string(data),
+	}, nil
+}
+
+// EnableService creates service file and enables it with systemctl enable
+func (sm *serviceManager) EnableService(service *Service) error {
+	// Write service file
+	err := ioutil.WriteFile(path.Join(sm.chroot, service.Path), []byte(service.Content), 0644)
+	if err != nil {
+		return err
+	}
+
+	// Change root dir
+	exit, err := utils.Chroot(sm.chroot)
+	if err != nil {
+		return err
+	}
+	defer exit()
+
+	// Enable service
+	cmd := exec.Command("systemctl", "enable", service.Name)
+	return cmd.Run()
+}
+
+// DisableService disables service with systemctl disable and removes service file
+func (sm *serviceManager) DisableService(service *Service) error {
+	// Disable service
+	exit, err := utils.Chroot(sm.chroot)
+	if err != nil {
+		return err
+	}
+	defer exit()
+
+	// Remove service file
+	cmd := exec.Command("systemctl", "enable", service.Name)
+	err = cmd.Run()
+	if err != nil {
+		return err
+	}
+
+	// Remove service file
+	err = os.Remove(service.Path)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// StartService starts given service on the system
+func (sm *serviceManager) StartService(service string) error {
+	exit, err := utils.Chroot(sm.chroot)
+	if err != nil {
+		return err
+	}
+	defer exit()
+
+	// Remove service file
+	cmd := exec.Command("systemctl", "start", service)
+	return cmd.Run()
+}
+
+// StopService starts given service on the system
+func (sm *serviceManager) StopService(service string) error {
+	exit, err := utils.Chroot(sm.chroot)
+	if err != nil {
+		return err
+	}
+	defer exit()
+
+	// Remove service file
+	cmd := exec.Command("systemctl", "stop", service)
+	return cmd.Run()
+}
+
+func (sm *serviceManager) isServiceLoaded(servicePath string) (bool, error) {
```

Comment:
What do you mean by loaded? 
Do you mean if the Service is activate? start? and why checking the file is good enough ?